tests/server/test_autoreloader.py

```diff
@@ -494,45 +528,119 @@ def test_main_some_args(self):
 
 
 class TestPollingReloader(unittest.TestCase):
-    class Stat(object):
-        # noinspection SpellCheckingInspection
-        def __init__(self, m, c):
-            self.st_mtime = m
-            self.st_ctime = c
+    def test_code_changed(self):
+        codec = codecs.lookup('utf8')
 
-    @mock.patch('pysoa.server.autoreload.os')
-    def test_code_changed(self, mock_os):
-        reloader = _PollingReloader('example_service.standalone', ['pysoa'])
+        with tempfile.NamedTemporaryFile('wb') as tmp_file1, tempfile.NamedTemporaryFile('wb') as tmp_file2, \
+                codecs.StreamReaderWriter(tmp_file1, codec.streamreader, codec.streamwriter, 'strict') as file1, \
+                codecs.StreamReaderWriter(tmp_file2, codec.streamreader, codec.streamwriter, 'strict') as file2:
+            reloader = _PollingReloader('example_service.standalone', ['pysoa'])
 
-        # noinspection PyUnresolvedReferences
-        with mock.patch.object(target=reloader, attribute='get_watch_file_names') as mock_get_watch_file_names:
-            mock_get_watch_file_names.return_value = [
-                '/path/to/first/file.py',
-                '/path/to/another/level.py',
-            ]
+            file1.write('test 1')
+            file1.flush()
 
-            mock_os.stat.side_effect = [self.Stat(123456789, 0), self.Stat(987654321, 0)]
+            file2.write('test 2')
+            file2.flush()
 
-            self.assertFalse(reloader.code_changed())
+            # noinspection PyUnresolvedReferences
+            with mock.patch.object(target=reloader, attribute='get_watch_file_names') as mock_get_watch_file_names:
+                mock_get_watch_file_names.return_value = [file1.name, file2.name]
 
-            self.assertEqual(2, mock_os.stat.call_count)
-            self.assertEqual('/path/to/first/file.py', mock_os.stat.call_args_list[0][0][0])
-            self.assertEqual('/path/to/another/level.py', mock_os.stat.call_args_list[1][0][0])
+                self.assertFalse(reloader.code_changed())
 
-            mock_os.reset_mock()
-            mock_os.stat.side_effect = [self.Stat(123456789, 0), self.Stat(987654321, 0)]
+                time.sleep(1.1)
```

Comment:
:-/

Reply:
Yeah, I wanted to test with actual temporary files instead of mocking `os.stat`. However, `st_mtime` is measured in seconds, so the only way to do this is to sleep.